mockOSApi.v2/Repositories/ProcessRepository.cs

```diff
@@ -3,51 +3,62 @@
 
 namespace mockOSApi.Repository;
 
-public interface IProcessRepository {
+public interface IProcessRepository
+{
     IEnumerable<MockProcess> GetAll();
     MockProcess? GetProcessByPid(int pid);
     Task CreateProcess();
     void KillProcess(MockProcess proc);
-  void ChangePriority(int prio,int pid);
+    void ChangePriority(int prio, int pid);
     public Task CreateProcess(MockProcess proc);
-     public Task Save();
+    public Task Save();
 };
 
-public class ProcessRepositoryDb: IProcessRepository {
+public class ProcessRepositoryDb : IProcessRepository
+{
 
     private readonly OSDbContext _dbContext;
-    public IEnumerable<MockProcess> GetAll() {
+    public IEnumerable<MockProcess> GetAll()
+    {
         return _dbContext.MockProcesses.ToList();
     }
-    public MockProcess? GetProcessByPid(int pid) {
+    public MockProcess? GetProcessByPid(int pid)
+    {
         return _dbContext.Find<MockProcess>(pid);
```

Comment:
```suggestion
        return _dbContext.MockProcesses.Where(p => p.pid == pid)
```
This is also natively supported by EF

Reply:
I can use this, but  _dbContext.MockProcesses.Where(p => p.pid == pid) returns an IQueryable<MockProcess>, which I don`t really need. Pid needs to be unique, so I only need to get back one entity, which _dbContext.Find<MockProcess> does already. With _dbContext.MockProcesses.Where(p => p.pid == pid) I will need to explicitly cast the IQueryable to a MockProcess?.